pkg/events/events.go

```diff
@@ -103,16 +103,40 @@ func New(object interface{}, eventType config.EventType, kind string) Event {
 		event.TimeStamp = obj.LastTimestamp.Time
 	case *apiV1.Pod:
 		event.Kind = "Pod"
+		if eventType == config.UpdateEvent {
+			condLen := len(obj.Status.Conditions)
+			if condLen != 0 {
+				event.TimeStamp = obj.Status.Conditions[condLen-1].LastTransitionTime.Time
+			}
+		}
 	case *apiV1.Node:
 		event.Kind = "Node"
+		if eventType == config.UpdateEvent {
```

Comment:
See this snippet being repeated across all conditions? Can we merge these somehow? See a lot of code duplication.

Reply:
To access resource obj, we need to typecast `runtime.Object` interface into resource object. That's what we do with switch cases. We can't access obj.Status from interface or without typecasting it